### PR TITLE
Bump Go and golang linter

### PR DESCRIPTION
**Pull request by milosgajdos**
Bump Go, golangci-lint and alpine versions.

**NOTE:**
* `vet` linter has been renamed to `govet`
* there is no longer a `run` section in the config
* `skip-dirs` has been renamed to `exclude-dirs`

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/distribution/distribution/v3
 
-go 1.21.8
+go 1.22.4
```
Comment (thaJeztah) [inline go.mod:3]: We can keep this on go1.21, unless the module can no longer work with go1.21 (which I doubt).

Comment (milosgajdos) [reply go.mod:3]: Kinda wanted to bump it in prep for beta release 🤔

Comment (thaJeztah) [reply go.mod:3]: right but the go mod now specifies the minimum _required_ version; trying to use any package from the repo will not be rejected unless you're on go1.22.4

We can use go1.22 for building the binaries, but not sure if we need to enforce that version for using the module

Comment (milosgajdos) [reply go.mod:3]: Yeah I'm familiar, I think we had a similar discussion in https://github.com/distribution/distribution/pull/4347 RE: go toolchain version

Comment (davidspek) [reply go.mod:3]: Then you can leave this line as is and add a line with `toolchain go`1.22.4`.

Comment (milosgajdos) [reply go.mod:3]: yah, I think @Jamstah and I discussed that but I cant remember what made us stick with the existing setup 🤷‍♂️ maybe he can chime in

Comment (davidspek) [reply go.mod:3]: Either way I think this should be merged :)

**Review by davidspek — APPROVED** [review]
Approved as is, or if the `toolchain` version is used instead. We should keep up with the Go versions.

File: go.mod (hunk 1 above)
Comment (Jamstah) [reply go.mod:3]: Go 1.21 will only be supported until ~August, so I feel like its probably worth bumping it.

Any project using this repo as a dependency should be keeping currency anyway. I don't think we break anyone until they update the dependency anyway.

**Review by Jamstah — APPROVED** [review]

**Pull request merged**